src/server/scripts/Kalimdor/zone_durotar.cpp

```diff
@@ -177,9 +178,571 @@ struct npc_mithaka : ScriptedAI
+            _pathCage = PATH_CAGE_TWO;
+            _pathHome = PATH_HOME_TWO;
+        }
+    }
+
+    void SetData(uint32 /*type*/, uint32 data) override
+    {
+        if (data == 1)
+        {
+            _eventInProgress = false;
+            me->SetGossipMenuId(GOSSIP_JAILOR_EVENT_READY);
+        }
+
+    }
+
+    void WaypointPathEnded(uint32 /*nodeId*/, uint32 pathId) override
+    {
+        if (pathId == _pathCage)
+        {
+            me->HandleEmoteCommand(EMOTE_ONESHOT_USE_STANDING);
+            me->CastSpell(me, SPELL_ACTIVATE_DNT);
+            _events.ScheduleEvent(EVENT_WALK_BACK_TO_HOME, 1s);
+        }
+        else if (pathId == _pathHome)
+        {
+            me->SetFacingTo(_facing);
+        }
+    }
+
+    bool OnGossipSelect(Player* player, uint32 /*menuId*/, uint32 /*gossipListId*/) override
+    {
+        player->KilledMonsterCredit(NPC_DARKSPEAR_JAILOR);
+        Talk(SAY_GET_IN_THE_PIT, player);
+        CloseGossipMenuFor(player);
+        _eventInProgress = true;
+        me->SetGossipMenuId(GOSSIP_JAILOR_EVENT_NOT_READY);
+        _events.ScheduleEvent(EVENT_PLAYER_ACCEPT_CHALLENGE, 2s);
+        return true;
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_PLAYER_ACCEPT_CHALLENGE:
+                    me->GetMotionMaster()->MovePath(_pathCage, false);
+                    break;
+                case EVENT_WALK_BACK_TO_HOME:
+                    if (Creature* scout = me->FindNearestCreature(NPC_CAPTIVE_SPITESCALE_SCOUT, 5.0f, true))
+                        scout->AI()->SetData(1, 1);
+                    me->GetMotionMaster()->MovePath(_pathHome, false);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        if (!UpdateVictim())
+            return;
+
+        DoMeleeAttackIfReady();
+    }
+private:
+    EventMap _events;
+    bool _eventInProgress;
+    uint32 _pathCage;
+    uint32 _pathHome;
+    float _facing;
+};
+
+enum CaptiveSpitescaleScout
+{
+    EVENT_UPDATE_JAILOR_GOSSIP    = 1,
+    EVENT_TALK_TO_PLAYER          = 2,
+    EVENT_DESPAWN_OUT_OF_COMBAT   = 3,
+    EVENT_CAST_FROSTBOLT          = 4,
+
+    POINT_PRISIONER_POSITION      = 0,
+
+    SAY_SSEND_YOU_TO_YOUR_DEATH   = 0,
+
+    SPELL_FROST_SHOCK             = 15089
+};
+
+Position const PRISIONER_POSITION_ONE = { -1142.49f, -5415.59f, 10.597655f };
+Position const PRISIONER_POSITION_TWO = { -1149.03f, -5526.18f, 8.1045685f };
+
+struct npc_captive_spitescale_scout : public ScriptedAI
+{
+    npc_captive_spitescale_scout(Creature* creature) : ScriptedAI(creature) { }
+
+    void JustAppeared() override
+    {
+        _events.ScheduleEvent(EVENT_UPDATE_JAILOR_GOSSIP, 2s);
+    }
+
+    void JustDied(Unit* /*killer*/) override
+    {
+        me->SetUnitFlag(UNIT_FLAG_IMMUNE_TO_PC);
```

Comment:
There's a precious SetImmuneToNPC existing

Reply:
It's there once.

Reply:
done